src/services/arweave/SimpleArweaveDataProvider.ts

```diff
@@ -98,6 +101,58 @@ export class SimpleArweaveDataProvider implements ArweaveDataProvider {
     });
   }
 
+  async validateArweaveAddress(address: string): Promise<undefined | boolean> {
+    const targetAddress = new ArweaveTransactionID(address);
+
+    const txPromise = this._arweave.api
+      .get(`/tx/${targetAddress.toString()}`)
+      .then((res: any) => (res.status === 200 ? res : false));
+
+    const balancePromise = this._arweave.api
+      .get(`/wallet/${targetAddress.toString()}/balance`)
+      .then((res: any) => (res.status === 200 ? res.data > 0 : false));
+
+    const gqlPromise = this._arweave.api
+      .post(`/graphql`, transactionByOwnerQuery(targetAddress))
+      .then((res: any) =>
+        res.status === 200 ? res.data.data.transactions.edges : false,
+      );
+
+    const [isTransaction, balance, hasTransactions] = await Promise.all([
+      txPromise,
+      balancePromise,
+      gqlPromise,
+    ]);
+
+    if (isTransaction) {
+      const tags = tagsToObject(isTransaction.data.tags);
+      const isContract = Object.values(tags).includes('SmartWeaveContract');
+      if (isContract) {
+        throw new Error(
+          `Provided address (${targetAddress.toString()}) is a SmartWeave Contract.`,
+        );
+      }
+      throw new Error(
+        `Provided address (${targetAddress.toString()} is a transaction ID and not a Smartweave Contract or an address.`,
+      );
+    }
+    if (balance) {
+      return true;
+    }
+    if (!balance) {
```

Comment:
ditch this check, just by getting here we know there is no `balance`

Reply:
removed

Reply:
maybe i wasn't clear. returning early is better. i.e. keep `if(balance) return true`. and get rid of this wrapped if statemnet.

Reply:
@atticusofsparta - please update ☝️

Reply:
updated in https://github.com/ar-io/pdns-react/pull/209/commits/982e3592adf3cab1fa9871675d369e4198b0ae80